docs/docs/reference-react-component.md

```diff
@@ -0,0 +1,259 @@
+
+### `props`
+
+`this.props` contains the props that were defined by the caller of this component. See [Components and Props](/react/docs/components-and-props.html) for an introduction to props.
+
+In particular, `this.props.children` is a special prop, typically defined by the child tags in the JSX expression rather than in the tag itself.
+
+### `propTypes`
+
+`propTypes` can be defined as a property on the component class itself, to define what types the props should be. It should be a map from prop names to types as defined in `React.PropTypes` In development mode, when an invalid value is provided for a prop, a warning will be shown in the JavaScript console. In production mode, `propTypes` checks are skipped for efficiency.
+
+For example, this code ensures that the `color` prop is a string:
+
+```js
+class CustomButton extends React.Component {
+  // ...
+}
+
+CustomButton.propTypes = {
+  name: React.PropTypes.string
+};
+```
+
+### `setState(nextState, callback)`
+
+Performs a shallow merge of nextState into current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
+
+The first argument can be an object (containing zero or more keys to update) or a function (of state and props) that returns an object containing keys to update.
+
+Here is the simple object usage:
+
+```javascript
+this.setState({mykey: 'my new value'});
+```
+
+It's also possible to pass a function with the signature `function(state, props) => newState` This can be useful in some cases when you want to enqueue an atomic update that consults the previous value of state and props before setting any values. For instance, suppose we wanted to increment a value in state:
+
+```javascript
+this.setState((prevState, currentProps) => {
+  return {myInteger: prevState.myInteger + 1};
+});
+```
+
+The second parameter is an optional callback function that will be executed once `setState` is completed and the component is re-rendered. Generally we recommend using `componentDidUpdate()` for such logic instead.
+
+`setState()` does not immediately mutate `this.state` but creates a pending state transition. Accessing `this.state` after calling this method can potentially return the existing value.
+
+There is no guarantee of synchronous operation of calls to `setState` and calls may be batched for performance gains.
+
+`setState()` will always lead to a re-render unless `shouldComponentUpdate()` returns `false` If mutable objects are being used and conditional rendering logic cannot be implemented in `shouldComponentUpdate()`, calling `setState()` only when the new state differs from the previous state will avoid unnecessary re-renders.
+
+### `state`
+
+The state contains data specific to this component that may change over time. The state is user-defined, and it should be a plain JavaScript object.
```

Comment:
It is worth mentioning that "If you don't use it in render() it shouldn't be in the state." Like you can put timer IDs on the instance.